src/markov/ddp.jl

```diff
@@ -0,0 +1,394 @@
+#=
+Discrete Decision Processes
+
+@author : Daisuke Oyama (Ported by Spencer Lyon, Matthew McKay)
+
+@date: 24/Sep/2015
+
+References
+----------
+
+http://quant-econ.net/jl/ddp.html
+
+Notes
+-----
+1. This currently implements ... Value Iteration, Policy Iteration, Modified Policy Iteration
+2. This does not currently support state-action pair formulation, or sparse matrices
+
+=#
+
+module DDP
+
+using QuantEcon
+
+import Base.*
+
+#-----------------#
+#-Data Structures-#
+#-----------------#
+
+"""
+DiscreteDP type for specifying paramters for discrete dynamic programming model
+
+##### Parameters
+
+- `Q::Array{T,NQ}` : Transition Probability Array
+- `R::Array{T,NR}` : Reward Array
+- `beta::Float64`  : Discount Factor
+
+##### Returns
+
+- `ddp::DiscreteDP` : DiscreteDP object
+
+"""
+type DiscreteDP{T<:Real,NQ,NR}
```

Comment:
`DiscreteDP(Q, R, beta)` versus `DiscreteDP(R, Q, beta)`: wouldn't it be better to have a common order between the Julia and Python versions?